README.md

```diff
@@ -1,4 +1,681 @@
+- 404: Unable to find the topic with topic name given by user.
+- 422: Semantically invalid request.
+- 503: An error occurred handling the request.
+
+Error responses carry a JSON body like the following:
+```json
+{"error_code":50301,"message":"Unknown Kafka Error", "incident_id": "17afe715-0ff5-4c49-9acc-a4204244a331"}
+```
+Error codes are of the format `HHHKK` where `HHH` is the HTTP Status Code and `KK` is the Kafka protocol error.  
+
+For E2E debugging purposes, the transaction ID of every request is returned in the HTTP header `X-Global-Transaction-Id`.
+If the header is set on the request, it will be honored. If not, it will be generated.
+In the event of a non-200 error return code, the transaction ID is also returned in the JSON error response as `incident_id`.
+
+
+## Using the REST API to administer Kafka topics
+
+The remainder of this document describes the Go-lang implementation of the Admin Rest SDK 
+and we also provide example `example.go` to show all SDK calls in action. 
+
+To run the example :-
+
+Compile the code.
+```sh
+	go build -o eventstreams-sdk-example
+```
+Set the required environment variables
+```sh
+# Set you API KEY.
+export API_KEY="abc123456789"
+
+# Set the Admin Endpoint to point to your cluster.
+export ADMIN_ENDPOINT="https://xyzclustername.svc01.region.eventstreams.test.cloud.ibm.com"
+
+```
+
+Run the example
+```sh
+./eventstreams-example 
+```
+
+## REST API 
+
+The following sections explain how the REST API works with examples.
+
+### Code Setup
+
+You will need to import the IBM SDK core functions and the rest API client itself.
+```golang
+	"github.com/IBM/go-sdk-core/v4/core"
+	"github.com/IBM/eventstreams-go-sdk/pkg/adminrestv1"
+```
+
+### Authentication
+
+Use one of the following methods to authenticate:
+
+- To authenticate using Basic Auth:
+  Use the user and api_key properties of the above objects as the username and password. Place these values into the Authorization header of the HTTP request in the form Basic <base64 encoding of username and password joined by a single colon (:)>.
+
+- To authenticate using a bearer token:
+  To obtain your token using the IBM Cloud CLI, first log in to IBM Cloud, then run the following command:
+  ```
+    ibmcloud iam oauth-tokens
+  ```
+    Place this token in the Authorization header of the HTTP request in the form Bearer. Both API key or JWT tokens are supported.
+
+- To authenticate directly using the api_key:
+  Place the key directly as the value of the X-Auth-Token HTTP header.
+
+```golang
+    URL := os.Getenv("ADMIN_ENDPOINT")
+	apiKey := os.Getenv("API_KEY")
+
+	if URL == "" {
+		fmt.Println("Please set env ADMIN_ENDPOINT")
+		os.Exit(1)
+	}
+
+	if apiKey == "" {
+		fmt.Println("Please set env API_KEY")
+		os.Exit(1)
+	}
+
+	basicAuthenticator, err := core.NewBasicAuthenticator("token", apiKey)
+	if err != nil {
+		fmt.Printf("Failed to Authenticate with error : %s\n", err.Error())
+		os.Exit(1)
+	}
+```
+
+
+### Create the service
```

Comment:
This title is confusing. It feels like we're going to create an Event Streams service instance. Instead we're creating a client for the Admin REST API

Reply:
### Creating a client for the Admin REST API.